### PR TITLE
Regression quickstart

### LINKED ISSUE #154: prediction error plot to support pandas Series object
When reading dataset as a pandas dataframe, the prediction error plot throws error `AttributeError: 'Series' object has no attribute 'reshape'`.
This code can solve the issue:
```
    if isinstance(y_true, pd.Series):
        y_reshaped = y_true.values.reshape((-1, 1))
    else:
        y_reshaped = y_true.reshape((-1, 1))
```

Additionally add test case for dataframe input.

### PR DESCRIPTION
**Pull request by neelasha23**
closes #154 #155 #159

### CODE REVIEW

File: docs/quickstart/regression.md
Comment (edublancas) [inline docs/quickstart/regression.md:16]: tip: the first 1-2 sentences should tell the reader what they will learn "in this tutorial, we'll use sklearn and sklearn-evaluation to fit and evaluate a regression model"

File: docs/quickstart/regression.md
Comment (edublancas) [inline docs/quickstart/regression.md:31]: what are the warnings that are displayed? we should address them instead of suppresing them

File: docs/quickstart/regression.md
Comment (edublancas) [inline docs/quickstart/regression.md:27]: please organize the [imports](https://peps.python.org/pep-0008/#imports):

```

    Standard library imports.
    Related third party imports.
    Local application/library specific imports.

```

File: docs/quickstart/regression.md
Comment (edublancas) [inline docs/quickstart/regression.md:41]: where is this data coming from? we need to cite the source and ensure the license allows us to use it. furthermore, it's better to download it from the original source instead of copying it and hosting it ourselves

File: docs/quickstart/regression.md
Comment (edublancas) [inline docs/quickstart/regression.md:57]: let's add some line breaks to improve readability

File: docs/quickstart/regression.md
Comment (edublancas) [inline docs/quickstart/regression.md:94]: `one measure for measuring` sounds redundant. maybe `R2 score measures how well...` try passing this through Grammarly to spot this problems

File: docs/quickstart/regression.md
Comment (edublancas) [inline docs/quickstart/regression.md:115]: maybe add some conclusion here based on the prediction error interpretation

File: docs/quickstart/regression.md
Comment (edublancas) [inline docs/quickstart/regression.md:118]: mention why removing outliers is important (they improve the model) and some rules of thumb to determine if removing a data point is sensible or not (I'd say that if the outliers are corrupted data makes sense, but if they are actual data points, we should not remove them, but check some resources and see what else they say)

File: src/sklearn_evaluation/plot/regression.py
Comment (edublancas) [inline src/sklearn_evaluation/plot/regression.py:126]: we can make this more readable:

```python
    if isinstance(y_true, pd.Series):
        y_reshaped = y_true.values

    y_reshaped = y_true.reshape((-1, 1))
```

File: src/sklearn_evaluation/plot/regression.py
Comment (edublancas) [inline src/sklearn_evaluation/plot/regression.py:129]: let's test with a few values to see if this yields good results

**Review by edublancas — CHANGES_REQUESTED** [review]

File: src/sklearn_evaluation/plot/regression.py
Comment (neelasha23) [reply src/sklearn_evaluation/plot/regression.py:128]: I haven't made this change as of now. Fixed the other comments. 
I have raised this [issue](https://github.com/ploomber/sklearn-evaluation/issues/159). 
I think this plot needs some rework. Should we take it up after merging the regression quickstart? 
Thoughts ? @edublancas

File: docs/quickstart/regression.md
Comment (neelasha23) [reply docs/quickstart/regression.md]: I guess it was redundant. No warnings in the code.

File: docs/quickstart/regression.md
Comment (neelasha23) [reply docs/quickstart/regression.md:27]: Moved imports to relevant cells.

File: docs/quickstart/regression.md
Comment (neelasha23) [reply docs/quickstart/regression.md:33]: Added reference. License: https://opendatacommons.org/licenses/dbcl/1-0/

File: docs/quickstart/regression.md
Comment (neelasha23) [reply docs/quickstart/regression.md]: The free version suggested some minor changes, I have incroporated those.

File: src/sklearn_evaluation/plot/regression.py
Comment (edublancas) [reply src/sklearn_evaluation/plot/regression.py:128]: I'll leave it up to you. If you think we need to fix the plot before merging this one, that's fine. Otherwise, we can merge this and then fix the plot. What do you think it's best? let me know so I know If I should do another pass to this PR or wait until the plot is fixed.

Comment (neelasha23) [reply src/sklearn_evaluation/plot/regression.py:128]: I think it's better to fix before the merge.

Comment (idomic) [reply src/sklearn_evaluation/plot/regression.py:128]: Please submit a PR?

File: docs/quickstart/regression.md
Comment (idomic) [reply docs/quickstart/regression.md:120]: @neelasha23 ^^

File: src/sklearn_evaluation/plot/regression.py
Comment (neelasha23) [inline src/sklearn_evaluation/plot/regression.py:121]: On analysing the code I realised that `model.fit(y_reshaped, y_pred)` is used for fitting the best fit line over the y_pred values. So this doesn't seem an issue. However, the `model` parameter seems unnecessary and might confuse the user. Since we just need a linear best fit line, `LinearRegression()` should suffice and can be the default. Other regressors like `Lasso()` or `Ridge()` are generating similar best fit lines as `LinearRegression()`. But this will cause the method signature to change. Thoughts? @edublancas @idomic

File: tests/test_regression.py
Comment (neelasha23) [inline tests/test_regression.py:61]: Added a couple of test cases to ensure the bounds for best fit line works.

File: src/sklearn_evaluation/plot/regression.py
Comment (idomic) [reply src/sklearn_evaluation/plot/regression.py:121]: So `Lasso/Ridge/LinearRegression(y_reshaped, y_pred)` instead of `model.fit(y_reshaped, y_pred)`?

Comment (edublancas) [reply src/sklearn_evaluation/plot/regression.py:121]: can you look up in the literature? see if there are situations where a user might want to fit a Lasso or Ridge? If we find some justification we can leave the argument there

Comment (neelasha23) [reply src/sklearn_evaluation/plot/regression.py:121]: I skimmed through some papers, didn't get exactly what we're looking for ( prediction error plot that also displays the `best fit` line). However below are some observations:
1. One common approach is to visualise the perpendicular distance of the error from the model's best fit line : 
error_plot

2. The closest that I could find was the 
[fit plot](https://www.sfu.ca/~mjbrydon/tutorials/BAinPy/09_regression.html#fit-plot) from `statsmodel`. However this doesn't visualise any `best fit` line, and shows the scatter plot only.

A few papers I referred to: (didn't find the prediction error plot with best fit line in these)
[On the interpretation of regression plots](https://sci-hub.se/https://www.tandfonline.com/doi/abs/10.1080/01621459.1994.10476459?journalCode=uasa20)
[Evaluating Random Forests for Survival Analysis using Prediction Error Curves](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4194196/)
[Modern approach to regression](https://d1wqtxts1xzle7.cloudfront.net/59391552/Sheather_-_A_Modern_Approach_to_Regression_with_R20190525-2191-a6v3gs-libre.pdf?1558803412=&response-content-disposition=inline%3B+filename%3DSpringer_Texts_in_Statistics.pdf&Expires=1672840294&Signature=KiJKPz0x~UUi~h3bpip6mdmIaWdCqs4HvEKPWXqVZkERnQazHQsY1H6dFpDfoDLLQ-9n~u-UnVY77wjsedqHaQ644queDSlVnId1BOWUJn9qf~vrKbK86MZ21vVIHRiQc8uq~i7nF-d8S4ZFr2FsIl6qbeDS9YJMqiKnJAAp5QSuIOpVQ9dH9LqJ7mfrPmk17iM87INLJkf6bULAEOEvoM7Ru7jDiBXV72Gyw4PWd8TMtxPxhlvzrYTSpT6dE-GGLsb8xvdzfQOYnhk1dqhBIL~xk5mv1MLnF12V0hFOXBpggiS4N1IIVq-eUJd4wmJPLt6fkZ6jtl2piBWjDWPlyg__&Key-Pair-Id=APKAJLOHF5GGSLRBV4ZA)

Comment (edublancas) [reply src/sklearn_evaluation/plot/regression.py:121]: ok, it makes sense to remove the model argument then. from what I understand, its only purpose is to serve as a benchmark to compare model predictions to the 45-degree line. let's deprecate it and remove it in an upcoming release. https://ploomber-core.readthedocs.io/en/latest/deprecation.html#parameter-deprecated

Comment (neelasha23) [reply src/sklearn_evaluation/plot/regression.py:131]: > ok, it makes sense to remove the model argument then. from what I understand, its only purpose is to serve as a benchmark to compare model predictions to the 45-degree line. let's deprecate it and remove it in an upcoming release. https://ploomber-core.readthedocs.io/en/latest/deprecation.html#parameter-deprecated

Yes that's right.
Added the deprecated directive.

File: src/sklearn_evaluation/plot/regression.py
Comment (edublancas) [inline src/sklearn_evaluation/plot/regression.py:115]: Please add a test for this (https://docs.pytest.org/en/7.1.x/how-to/capture-warnings.html#warns)

also, the deprecation will happen in two major versions to that'd be `0.10`, I think since we recently release this, let's remove it in 0.9

**Review by edublancas — CHANGES_REQUESTED** [review]

File: src/sklearn_evaluation/plot/regression.py
Comment (idomic) [reply src/sklearn_evaluation/plot/regression.py:115]: + We've added links to the `intro.md` page, include this guide as well in it.

Comment (edublancas) [reply src/sklearn_evaluation/plot/regression.py:115]: deprecation is fine 0.8.6, removal in 0.9

Comment (neelasha23) [reply src/sklearn_evaluation/plot/regression.py:115]: Added in intro.md @idomic

**Review by edublancas — APPROVED** [review]

**Comment by edublancas** [comment]
merged, please check the live docs in a few mins to see if all looks good! @neelasha23

**Comment by neelasha23** [comment]
looking good!

**Comment by idomic** [comment]
Nice job!

**Pull request merged**
